cmd/orchestrate/orchestrate.go

```diff
@@ -0,0 +1,318 @@
+	AccessionID        string      `json:"accession_id"`
+	DecryptedChecksums []checksums `json:"decrypted_checksums"`
+}
+
+type mapping struct {
+	Type         string   `json:"type"`
+	DatasetID    string   `json:"dataset_id"`
+	AccessionIDs []string `json:"accession_ids"`
+}
+
+// Checksums is struct for the checksum type and value
+type checksums struct {
+	Type  string `json:"type"`
+	Value string `json:"value"`
+}
+
+func main() {
+	conf, err := config.NewConfig("orchestrate")
+	if err != nil {
+		log.Fatal(err)
+	}
+	mq, err := broker.NewMQ(conf.Broker)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer mq.Channel.Close()
+	defer mq.Connection.Close()
+
+	queues := []string{queueInbox, queueVerify, queueComplete}
+
+	go func() {
+		connError := mq.ConnectionWatcher()
+		log.Error(connError)
+		os.Exit(1)
+	}()
+
+	routing := map[string]string{
+		queueVerify:   queueAccession,
+		queueInbox:    queueIngest,
+		queueComplete: queueMapping,
+	}
+
+	forever := make(chan bool)
+
+	log.Info("Starting orchestrate service")
+
+	for _, queue := range queues {
+		routingKey := routing[queue]
+		go processQueue(mq, queue, routingKey, conf)
+	}
+	<-forever
+}
+
+func processQueue(mq *broker.AMQPBroker, queue string, routingKey string, conf *config.Config) {
+	durable := conf.Broker.Durable
+
+	log.Infof("Monitoring queue: %s", queue)
+
+	messages, err := mq.GetMessages(queue)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for delivered := range messages {
+		log.Debugf("Received a message: %s", delivered.Body)
+
+		schema, err := schemaNameFromQueue(queue, delivered.Body)
+
+		if err != nil {
+			log.Errorf("Don't know schema for message type "+
+				"(corr-id: %s, queue: %s, error: %v, message: %s)",
+				delivered.CorrelationId,
+				queue,
+				err,
+				delivered.Body)
+
+			continue
+		}
+
+		err = mq.ValidateJSON(&delivered, schema, delivered.Body, nil)
+
+		if err != nil {
+			log.Errorf("Validation failed for message "+
+				"(corr-id: %s, error: %v, schema: %s, message: %s)",
+				delivered.CorrelationId,
+				err,
+				schema,
+				delivered.Body)
+
+			continue
+		}
+
+		var publishMsg []byte
+		var publishType interface{}
+
+		routingSchema, err := schemaNameFromQueue(routingKey, nil)
+
+		if err != nil {
+			log.Errorf("Don't know schema for routing key "+
+				"(corr-id: %s, routing-key: %s, error: %v)",
+				delivered.CorrelationId,
+				routingKey,
+				err,
+			)
+
+			continue
+		}
+
+		switch routingKey {
+		case queueAccession:
+			publishMsg, publishType = finalizeMessage(delivered.Body, conf)
+		case queueIngest:
+			publishMsg, publishType = ingestMessage(delivered.Body)
+		case queueMapping:
+			publishMsg, publishType = mappingMessage(delivered.Body, conf)
+		}
+
+		err = mq.ValidateJSON(&delivered,
+			routingSchema,
+			publishMsg,
+			publishType)
+
+		if err != nil {
+			log.Errorf("Validation of outgoing message failed "+
+				"(corr-id: %s, "+
+				"error: %v)",
+				delivered.CorrelationId,
+				err)
+
+			continue
+		}
```

Comment:
```suggestion
		if err != nil {
			log.Errorf("Validation of outgoing message failed, error: %v", err)
			if err := delivered.Nack(true); err != nil {
				log.Errorf("failed to nack message for reason: %v", err)
			}

			continue
		}
```

If this where to happen for some strange reason we need to `Nack` and re-queue the original message.

Reply:
I'm assuming the `Nack()` call should have been `Nack(false, true)`.